### PR TITLE
feat(core): gather task inputs and build a hash plan

### PR DESCRIPTION
**Pull request by Cammisuli**
## Current Behavior

Its hard to determine why a task was affected by certain changes. If there are a lot of inputs + their dependencies, it can get overwhelming to mentally map the inputs.

## Expected Behavior

The "task plan" will now be built separately from hashing, and we can check what inputs actually affect our targets/tasks. The UI portion of this will come in a separate PR

## Related Issue(s)

Fixes #

TODO:
- [x] Add "task plan" information to be consumed for the graph view

### CODE REVIEW

File: packages/nx/src/tasks-runner/run-command.ts [hunk 1]
```diff
@@ -107,8 +108,9 @@ function createTaskGraphAndValidateCycles(
     loadDotEnvFiles: boolean;
   }
 ) {
-  const taskGraph = createTaskGraph(
+  const taskGraph = createTaskGraphWithPlan(
```
Comment (FrozenPandaz) [inline packages/nx/src/tasks-runner/run-command.ts]: Remove these changes for now.

File: packages/nx/src/tasks-runner/create-task-graph.ts [hunk 2]
```diff
@@ -398,6 +403,38 @@ export function mapTargetDefaultsToDependencies(
   return res;
 }
 
+export function createTaskGraphWithPlan(
```
Comment (FrozenPandaz) [inline packages/nx/src/tasks-runner/create-task-graph.ts:406]: This should be a function like..

```ts
createHashPlans(
  nxJson,
  projectGraph,
  taskGraph
): HashPlans
```

File: packages/nx/src/tasks-runner/create-task-graph.ts [hunk 3]
```diff
@@ -398,6 +403,38 @@ export function mapTargetDefaultsToDependencies(
...
+  projectGraph: ProjectGraph,
+  nxJson: NxJsonConfiguration,
+  defaultDependencyConfigs: TargetDependencies,
+  projectNames: string[],
+  targets: string[],
+  configuration: string | undefined,
+  overrides: Object,
+  excludeTaskDependencies: boolean = false
+) {
+  const taskGraph = createTaskGraph(
```
Comment (FrozenPandaz) [inline packages/nx/src/tasks-runner/create-task-graph.ts:416]: Don't create the task graph in this function, take it as an argument.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Comment by Cammisuli** [comment]
Closing this as this has been implemented again with #19233

**Pull request closed without merging**